File: backend/core/core-domain/src/main/java/com/ssafy/s12p21d206/achu/domain/Goods.java
package com.ssafy.s12p21d206.achu.domain;

import com.ssafy.s12p21d206.achu.domain.support.DefaultDateTime;
import java.util.Objects;

public record Goods(
    Long id,
    String title,
    String description,
    ImageUrlsWithThumbnail imageUrlsWithThumbnail,
    TradeStatus tradeStatus,
    Long price,
    DefaultDateTime defaultDateTime,
    Long categoryId,
    User user,
    Long babyId) {

  public boolean isSamePrice(Long price) {
    return Objects.equals(this.price, price);
  }
}

File: android/app/src/main/java/com/ssafy/achu/data/model/chat/Goods.kt
package com.ssafy.achu.data.model.chat

import kotlinx.serialization.Serializable

@Serializable
data class Goods(
    val id: Int,
    val thumbnailImageUrl: String,
    val title: String,
    val price: Long,
    val tradeStatus: String,
)